### ISSUE TITLE
Fails to unmarshall free-form objects (additionalProperties)

### ISSUE THREAD

**Issue by pcrespov** [p1c2u/openapi-core#153]
Current version can validate free-form objects but **cannot** unmarshall them.

A free-form object (arbitrary property/value pairs) is defined equivalently [see reference](https://swagger.io/docs/specification/data-models/data-types/) as
```yaml
type: object
```
or
```yaml
type: object
additionalProperties: true
```
or
```yaml
type: object
additionalProperties: {}
```
PR #121 address a related issue that solves validation but still unable to unmarshall. For instance, the [example](https://github.com/p1c2u/openapi-core/issues/67#issuecomment-463656001) proposed by @berend in the same pull request cannot be unmarshalled as demonstrated in a [tests I did in my fork](https://github.com/pcrespov/openapi-core/commit/c69cd04f170ccbe664994d45b6bc14b97abf8d40)

Any hint on how to overcome/fix this?

**Comment by pcrespov** [comment]
Any hint on how to overcome this?

**Comment by playpauseandstop** [comment]
Hi,

While working on adding OpenAPI 3 support for `aiohttp.web` applications I ran into same issue. 

I managed to fix it ugly by merging [casted data into unmarshaled data](https://github.com/playpauseandstop/rororo/blob/d35010d210d00a8529a91439d2540563e669159d/rororo/openapi/validators.py#L32), but I wonder maybe there are plans on support free-form objects in `openapi-core`?

----

I understand that `Model` class based on top of list of schema [properties](https://github.com/p1c2u/openapi-core/blob/c3f9adadaa82223600369b53919530baf8795930/openapi_core/extensions/models/models.py#L15), but there should be an exception for next OpenAPI 3 schema,

```yaml
  type: "object"
  additionalProperties: true
```

And even more. As [additionalProperties](https://spec.openapis.org/oas/v3.0.2#properties) is `true` by default,

> additionalProperties - Value can be boolean or object. Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema. Consistent with JSON Schema, additionalProperties defaults to true.

maybe it is needed to redesign `Model` to support, by default, any properties sent to the object and only in case of,

```yaml
  type: "object"
  properties:
    one:
       type: "string"
    two:
       type: "string"
  additionalProperties: false
```

enforce current behavior?
